web/js/precallView.js

```diff
@@ -0,0 +1,236 @@
+/* globals setTimeout */
+!(function (exports) {
+  'use strict';
+
+  var _templateSrc = '/templates/precall.ejs';
+  var _template;
+  var _model;
+  var testMeterInterval;
+
+  var addHandlers = function () {
+    var preCallTestResults = document.getElementById('pre-call-test-results');
+
+    preCallTestResults.addEventListener('click', function (e) {
+      var elem = e.target;
+      switch (elem.id) {
+        case 'precall-close':
+          preCallTestResults.style.display = 'none';
+          break;
+        case 'retest':
+          preCallTestResults.style.display = 'none';
+          Utils.sendEvent('roomView:retest');
+          break;
+      }
+    });
+
+    var publishSettings = document.querySelector('.publish-settings');
+
+    publishSettings.addEventListener('click', function (e) {
+      var initialVideoSwitch = document.querySelector('#initialVideoSwitch');
+      var initialAudioSwitch = document.querySelector('#initialAudioSwitch');
+      var elem = e.target;
+      elem.blur();
+      // pointer-events is not working on IE so we can receive as target a child
+      elem = HTMLElems.getAncestorByTagName(elem, 'a');
+      if (!elem) {
+        return;
+      }
+      switch (elem.id) {
+        case 'initialAudioSwitch':
+          if (!initialAudioSwitch.classList.contains('activated')) {
+            setSwitchStatus(true, true, 'Audio', 'roomView:initialAudioSwitch');
+          } else {
+            setSwitchStatus(false, true, 'Audio', 'roomView:initialAudioSwitch');
+          }
+          break;
+        case 'initialVideoSwitch':
+          if (!initialVideoSwitch.classList.contains('activated')) {
+            setSwitchStatus(true, true, 'Video', 'roomView:initialVideoSwitch');
+          } else {
+            setSwitchStatus(false, true, 'Video', 'roomView:initialVideoSwitch');
+          }
+          break;
+      }
+    });
+
+    var videoPreviewElement = document.getElementById('video-preview');
+    var videoPreviewNameElement = document.getElementById('video-preview-name');
+    videoPreviewElement.addEventListener('mouseover', function () {
+      videoPreviewNameElement.style.opacity = 1;
```

Comment:
I think this would be better in the css using `:hover`

Reply:
@maikthomas, The`mouseover` target (`videoPreviewElement`) is different than the element for which we are setting the opacity (`videoPreviewNameElement`). So, CSS `:hover` won't work.

Reply:
Hmm ok, so the alternative would be to set a data-attr or add a class but we would still need this code so not worth much. Thanks for clarifying.